xbmc/windowing/X11/GLContextEGL.cpp

```diff
@@ -205,7 +205,7 @@ bool CGLContextEGL::CreatePB()
     EGL_BLUE_SIZE, 8,
     EGL_GREEN_SIZE, 8,
     EGL_RED_SIZE, 8,
-    EGL_DEPTH_SIZE, 8,
+    EGL_DEPTH_SIZE, 24,
```

Comment:
We request 16 bits depth buffer on most (all?) other platforms. Is that not enough? I'm not sure all (embedded) devices can support 24. @lrusak ?

Reply:
Not sure, seen only on all other places a used "24" (also a bit below in this file) and seen this "8" as c&p from line above.

Reply:
@FernetMenta better to change this and the "if (!eglGetConfigAttrib(eglDisplay, config, EGL_DEPTH_SIZE, &value) || value < 24)" to 16?

Reply:
The pixel buffer, which is used for offscreen rendering, does not need this modification.

Regarding normal rendering, there is nothing in the Kodi application that requires a depth buffer. I won't waste resources just for the odd case of a fancy screen saver that does depth testing. Addons could request special features, like a depth buffer, on demand. That saves overhead not needed in general.

Reply:
@FernetMenta In principle, you are of course right - Kodi does not need a depth buffer. But on-demand depth buffer requests will (at least I assume so) not go into v18 since the changes that would be required are too far-reaching. Right now, the behavior we have is a regression compared to v17. A number of visualizations and screensavers that used to work are now broken due to the depth buffer being absent.

As I don't think requesting the unused depth buffer will incur more overhead than the space it occupies in the GPU RAM and we were fine having it until now, I'd keep it for the time being. When we would make the depth buffer available on demand in the future, do you have any insight into how much the GLX/EGL surface recreation would const, i.e. would it possibly stutter for example?

As a side-note, `GLContextGLX` also requests 24 bits of depth buffer currently.